### PR TITLE
Add SqliteTypedQueryBuilder for exact type mapping

### PR DESCRIPTION
**Pull request by Perksey**
## PR Info

- Related to (doesn’t necessarily close) #689 

## New Features

Adds customizable type mapping to ensure that the exact type can be used on SQLite if the user needs it. 

## Bug Fixes

Provides the ability for the user to enable BigUnsigned PRIMARY KEY AUTOINCREMENT.

## Breaking Changes

I’ve done my best to avoid any breaking changes. This is done by introducing a SqliteTypedQueryBuilder and implementing a SqliteBuilderVariant trait on both builders to deduplicate the code.

## Changes

I was hesitant to add a blanket implementation of sea-query types on T where T is a SqliteBuilderVariant, as this would definitely cause future problems, either here or downstream. Therefore, I’ve implemented a sqlite_impl macro which will repeat the implementation for both SqliteTypedQueryBuilder<T> and SqliteQueryBuilder.

I forgot about the need of PhantomData when I originally proposed this, and this can’t be done in a non-breaking manor without separate types.

### CODE REVIEW

**Comment by Perksey** [comment]
This can close #689 but discussion is needed for whether we actually want to special case BIGINT AUTOINCREMENT on the default mapper.

**Comment by Perksey** [comment]
@tyt2y3 ready for review

File: src/backend/sqlite/table.rs [hunk 1]
```diff
@@ -1,48 +1,168 @@
...
 
-    fn prepare_column_type(&self, column_type: &ColumnType, sql: &mut dyn SqlWriter) {
+impl StaticTypeMapper for DefaultTypeMapper {
+    fn prepare_column_type(column_type: &ColumnType, sql: &mut dyn SqlWriter) {
         write!(
             sql,
             "{}",
             match column_type {
+                // ----------------------------------------------------------------------------------------------------
+                // A NOTE REGRADING SQLITE TYPE AFFINITY: It's great that we try to be very on-point with our mappings,
```
Comment (tyt2y3) [inline src/backend/sqlite/table.rs:136]: That's a very throughout comment section haha

**Review by tyt2y3 — COMMENTED** [review]
It looks pretty good.

My only hope is to have a prettier macro that can do the following:

```rust
#[impl_many]
impl T for A, B {
    fn method() {}
}
```

that would create no doubt about what the `sqlite_impl` would do

**Comment by Perksey** [comment]
I couldn't find any macros that would support adding a generic parameter for one of the alternate selves, and attempting to make one did not go well!

**Comment by tyt2y3** [comment]
Alternatively, may be we can implement only a subset of the traits?

**Pull request closed without merging**